leader/undirected_graph/sync_casteigts_metivier_robson_zemmari/sync_casteigts_metivier_robson_zemmari.go

```diff
@@ -0,0 +1,551 @@
+package sync_casteigts_metivier_robson_zemmari
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/bits"
+	"math/rand"
+	"time"
+
+	"github.com/krzysztof-turowski/distributed-framework/lib"
+)
+
+// Run the Casteigts-Métivier-Robson-Zemmari deterministic leader election algorithm on a random network
+func Run(n int, p float64) (int, int) {
+	nodes, synchronizer := buildSynchronizedRandomConnectedGraph(n, p)
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	identifiers := rng.Perm(n)
+	for i, node := range nodes {
+		log.Println("Node", node.GetIndex(), "about to run")
+		go run(node, 2*(identifiers[i]+1)) // Every node is assigned a unique identifier of O(log(n)) bits
```

Comment:
And what if there are some duplicates? (IIRC that happens with constant probability) Maybe it's better to introduce a helper function for writing unique ID to nodes.